### PR TITLE
🐛 Switch analytics video play event to video play event.

### LINKED ISSUE #23509: Use play event for analytics instead of playing
Related to https://github.com/ampproject/amphtml/pull/23317#issuecomment-514438498

The `playing` event fires when a video begins playing for any reason, including after the buffer runs out and the video must pause then play again. `play` fires when the user initiates a play event. It looks like a mistake that the `playing` event was used initially.

| Event | Description |
|--|--|
|`play`|The element is no longer paused. Fired after the play() method has returned, or when the autoplay attribute has caused playback to begin.|
|`playing`|Playback is ready to start after having been paused or delayed due to lack of media data.|

Based on this, the event should be `play` if we want to track when the user plays the video.

- [ ] Somehow communicate that metrics will change. This could include updating the documentation or messaging the slack channel or the changelog for the release.
- [ ] Change the event listener from `playing` to `play`

/to @zhouyx for priority and triage
/cc @wassgha

### PR DESCRIPTION
**Pull request by cramforce**
For backward compat, because not all video players emit a play event, this change continues to emit the analytics play upon the video playing event if no video play event has been seen.

The existing integration tests should cover this as there isn't a change in behavior for simple cases (where playback isn't interrupted by network issues).

Fixes #23509

### CODE REVIEW

File: src/service/video-manager-impl.js [hunk 1]
```diff
@@ -502,6 +505,10 @@ class VideoEntry {
       this.videoLoaded()
     );
     listen(video.element, VideoEvents.PAUSE, () => this.videoPaused_());
+    listen(video.element, VideoEvents.PLAY, () => {
```
Comment (zhouyx) [inline src/service/video-manager-impl.js:508]: I asked this question offline. `play` event should be strictly before `playing` event, but can we guarantee this if the event are passed via postMessages.

**Comment by cramforce** [comment]
@alanorozco Very friendly review ping 🙏🏾

**Comment by cramforce** [comment]
@zhouyx What would be the right place to document this?

**Comment by zhouyx** [comment]
Here's the video analytics doc https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/amp-video-analytics.md#video-play-trigger-on-video-play

**Comment by cramforce** [comment]
Added docs.

**Pull request merged**